### ISSUE TITLE
For vue-apollo@3 and apollo-client@3, an old `apollo-client@2` is still required.

### ISSUE THREAD

**Issue by mstys** [vuejs/vue-apollo#1059]
**Describe the bug**
After upgrading from apollo-v2 to apollo-v3 and vue-apollo from v2.* to v3.*  vue-apollo still requries old dependencies from `apollo-client` package.

**To Reproduce**
1. Upgrade project following an official tutorial: https://www.apollographql.com/docs/react/migrating/apollo-client-3-migration/#updating-imports
2. Delete unneeded `apollo-client` library.
2. Run project.
3. Get err.. 
```
node_modules/vue-apollo/types/vue-apollo.d.ts
13:28 Cannot find module 'apollo-client/util/Observable' or its corresponding type declarations.
    11 | } from 'apollo-client'
    12 | import { FetchResult } from 'apollo-link'
  > 13 | import { Observable } from 'apollo-client/util/Observable'
       |                            ^
    14 | import { ApolloProvider, VueApolloComponent } from './apollo-provider'
    15 | import {
    16 |   VueApolloQueryDefinition,
```

**Expected behavior**
Replace imports an old dependencies from `apollo-client` with new imports from`@apollo/client`.

**Versions**
```
  "vue-apollo": "^3.0.4",
  "@apollo/client": "^3.2.0",
  "vue": "^2.6.12",
```

**Comment by mstys** [comment]
**Update issue**
For typescript user
```ts
ERROR: Type 'ApolloClient<any>' is missing the following properties from type 'ApolloClient<any>': store, writeData, initQueryManager
Type import ("/node_modules/@apollo/client/core/ApolloClient").ApolloClient<any>' is not assignable to type 'import("/node_modules/apollo-client/ApolloClient").default<any>'.
   > 180 |   defaultClient: apolloClient,
         |   ^
```

```ts 
// file: apollo-provider.d.ts

import Vue, { AsyncComponent } from 'vue'
import { ApolloClient } from 'apollo-client'.   // <-------
import {
  VueApolloComponentOptions,
  WatchLoading,
  ErrorHandler
} from './options'

export type VueApolloComponent<V extends Vue = Vue> = VueApolloComponentOptions<V> | typeof Vue | AsyncComponent

export class ApolloProvider<TCacheShape=any> {
  provide: (key?: string) => this
  constructor (options: {
    defaultClient: ApolloClient<TCacheShape>,   // <--------
    defaultOptions?: VueApolloComponentOptions<Vue>,
    clients?: { [key: string]: ApolloClient<TCacheShape> },
    watchLoading?: WatchLoading,
    errorHandler?: ErrorHandler,
    prefetch?: boolean
  })
  clients: { [key: string]: ApolloClient<TCacheShape> }
  defaultClient: ApolloClient<TCacheShape>
}
```

so types for ApolloProvider should be updated as well.

**Comment by Akryum** [comment]
Fixed in 6dc7c8ee2c03adcb8abcff38067d2eaece47b658

**Comment by quentinus95** [comment]
@Akryum I don't understand why this ticket is marked as fixed, am I missing something?

- https://github.com/vuejs/vue-apollo/blob/v3.0.5/packages/vue-apollo/types/apollo-provider.d.ts#L4
- https://github.com/vuejs/vue-apollo/blob/v3.0.5/packages/vue-apollo/types/vue-apollo.d.ts

Those files still rely on `apollo-client` (Apollo 2.x).

How can an existing project using VueApollo 3.x and @vue/apollo-composable can update to Apollo Client 3?

**Comment by sandorTuranszky** [comment]
it's 6th of Jab 2021 and the isses is still there. Vue-apollo version 3.0.5

**Comment by TQ-Benji** [comment]
30th June 2021 - Still not fixed on 3.0.7

**Comment by HansCronau** [comment]
Note that all [Vue Apollo 3 documentation] still instructs to install `apollo-client` 2 as a peer dependency instead of `@apollo/client` 3. It seems this will change starting [Vue Apollo 4].

[Vue Apollo 3 documentation]: https://apollo.vuejs.org/guide/installation.html
[Vue Apollo 4]: https://v4.apollo.vuejs.org/guide/installation.html
